docs/api.rst

```diff
@@ -4689,41 +4689,170 @@ Dealing with History Events
+         :reqjson string[optional] location_label: location_label is a string field that allows to provide more information about the location. For example when we use this structure in blockchains can be used to specify the source address.
+         :reqjson string[optional] notes: This is a description of the event entry in plain text explaining what is being done. This is supposed to be shown to the user.
+         :reqjson string[optional] counterparty: An identifier for a potential counterparty of the event entry. For a send it's the target. For a receive it's the sender. For bridged transfer it's the bridge's network identifier. For a protocol interaction it's the protocol.
+         :reqjson object[optional] extra_data: An object containing any other data to be stored.
+
+   .. tab:: History Event
+
+      **Example Request**:
+
+      .. http:put:: /api/(version)/history/events
+
+         .. http:example:: curl wget httpie python-requests
+
+            PUT /api/1/history/events HTTP/1.1
+            Host: localhost:5042
+            Content-Type: application/json;charset=UTF-8
+
+            {
+               "entry_type": "history event",
+               "event_identifier": "RE_xxxxxxxxxx",
+               "location": "ethereum",
+               "timestamp": 1569924574,
+               "balance": {"amount": "1.542", "usd_value": "1.675"},
+               "sequence_index": 162,
+               "event_type": "informational",
+               "event_subtype": "approve",
+               "asset": "eip155:1/erc20:0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359",
+               "location_label": "0x2B888954421b424C5D3D9Ce9bB67c9bD47537d12",
+               "notes": "Approve 1 SAI of 0x2B888954421b424C5D3D9Ce9bB67c9bD47537d12 for spending by 0xdf869FAD6dB91f437B59F1EdEFab319493D4C4cE"
+            }
+
+         :reqjson int sequence_index: This is an index that tries to provide the order of history entries for a single event_identifier.
+         :reqjson string location: The location of the entry. Such as "ethereum", "optimism", etc.
+         :reqjson string asset: The asset identifier for this entry
+         :reqjson string event_identifier: The event identifier to be used for the event.
+         :reqjson string event_type: The main event type of the entry. Possible event types can be seen in the `HistoryEventType enum <https://github.com/rotki/rotki/blob/59aa288dacd1776e62682e711a916f32a14c04c2/rotkehlchen/accounting/structures/types.py#L54>`_.
+         :reqjson string[optional] event_subtype: An optional subtype for the entry. Possible event types can be seen in the `HistoryEventSubType enum <https://github.com/rotki/rotki/blob/59aa288dacd1776e62682e711a916f32a14c04c2/rotkehlchen/accounting/structures/types.py#L72>`_.
```

Comment:
Same here. Why are you duplicating those btw? Shouldn't they be in the base description of the hisory event only?

```suggestion
         :reqjson event_subtype: The subtype for the entry. Possible event types can be seen in the `HistoryEventSubType enum <https://github.com/rotki/rotki/blob/59aa288dacd1776e62682e711a916f32a14c04c2/rotkehlchen/accounting/structures/types.py#L72>`_.
```

Reply:
they only appear in two of the subclasses. `entry_type`, `timestamp` & `balance` are the only fields that apply to all subclasses